spark/src/main/scala/org/apache/spark/sql/delta/commands/merge/MergeIntoMaterializeSource.scala

```diff
@@ -300,12 +301,12 @@ trait MergeIntoMaterializeSource extends DeltaLogging with DeltaSparkPlanUtils {
       assert(rdd.isCheckpointed)
     }
 
-    logDebug(s"Materializing MERGE with pruned columns $referencedSourceColumns. ")
-    logDebug(s"Materialized MERGE source plan:\n${sourceDF.get.queryExecution}")
+    logDebug(s"Materializing MERGE with pruned columns $referencedSourceColumns.")
+    logDebug(s"Materialized MERGE source plan:\n${getSourceDF.queryExecution}")
     materializeReason
   }
 
-  protected def getSourceDF(): Dataset[Row] = {
+  protected def getSourceDF: DataFrame = {
```

Comment:
nit: I would keep the empty param list here has the name suggests it's a method more than an an attribute.

Reply:
What do you think about [this](https://docs.scala-lang.org/glossary/index.html#uniform-access-principle)?

> The uniform access principle states that variables and parameterless functions should be accessed using the same syntax. Scala supports this principle by not allowing parentheses to be placed at call sites of parameterless functions. As a result, a parameterless function definition can be changed to a val, or vice versa, without affecting client code.

I think this method is actually a perfect example of a pure accessor and hence no need for `()`.

I think the only reason why this method got `()` is the Java-like accessor name.

Just sharing my 0.02 cents here yet changing the code to use `()` as I'm actually OK either way.

Reply:
I'm fine with both also, if we remove the parentheses I would just call it `sourceDF` tbh and find a different name for the internal variable but not sure that's worth the effort at this point